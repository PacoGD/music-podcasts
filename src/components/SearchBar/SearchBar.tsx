import { FunctionComponent, useState } from "react";
import styles from "./style.module.css";

export const SearchBar: FunctionComponent = () => {
  const [searchQuery, setSearchQuery] = useState<string>("");

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchQuery(e.target.value);
  };

  return (
    <div className={styles.container}>
      <p className={styles.searches}>100</p>
      <input
        type="text"
        className={styles.input}
        value={searchQuery}
        onChange={handleChange}
        placeholder="Filter podcasts..."
      />
    </div>
  );
};
