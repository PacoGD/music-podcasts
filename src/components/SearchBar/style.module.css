.searches {
  color: #fff;
  font-size: 12px;
  background-color: #66b0b8;
  padding: 0 5px 0 5px;
  font-weight: 600;
  border-radius: 7px;
  margin: 0 3px 0 0;
}

.container {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
}

.input {
  font-size: 10px;
  width: 180px;
  height: 15px;
  border-radius: 3px;
  padding: 3px 12px 3px 12px;
  margin: 10px 40px 10px 7px;
  border: 3px;
  box-shadow: 0 1px 3px #999;
}

.input::placeholder {
  color: #999; /* Cambia el color a gris, por ejemplo */
}
