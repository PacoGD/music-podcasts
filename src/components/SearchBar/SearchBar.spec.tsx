import { render } from "@testing-library/react";
import { SearchBar } from "./SearchBar";

describe("SearchBar component", () => {
  const onChange = jest.fn();
  const value = "";
  const filteredPodcastCount = 0;
  test(" snapshot", () => {
    const { container } = render(
      <SearchBar
        onChange={onChange}
        value={value}
        filteredPodcastsCount={filteredPodcastCount}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
