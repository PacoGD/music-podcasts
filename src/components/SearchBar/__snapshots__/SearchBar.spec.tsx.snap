// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component  snapshot 1`] = `
<div>
  <div>
    <p>
      0
    </p>
    <input
      placeholder="Filter podcasts..."
      type="text"
      value=""
    />
  </div>
</div>
`;
